### PR TITLE
discriptive error alerts and token improvements

### PR DESCRIPTION
**Pull request by OttoAhlfors**
I seem to have accidentaly forgotten to change the branch for improving token handling so its here now, suprise!

### CODE REVIEW

**Comment by tuomasmustakallio** [comment]
Could the PR title be changed if this also contains the token ticket?

File: client/components/searchComponents/search.tsx [hunk 1]
```diff
@@ -51,22 +49,24 @@ const Search: React.FC<SearchProps> = ({
...
+                headers: { Authorization: localStorage.getItem('token') },
             })
             .then((response) => {
                 updatePool(response.data)
                 enableAddButton()
             })
             .catch((error) => {
                 setErrorAlert(
-                    `Joining to pool failed with error: ${error.message}`,
+                    `Joining to pool failed with error : ${error.response.data.detail}`,
```
Comment (EddieTheCubeHead) [inline client/components/searchComponents/search.tsx:69]: "Joinin **a** pool"

File: client/components/poolmanagerComponents/poolManager.tsx [hunk 2]
```diff
@@ -30,15 +28,15 @@ const PoolManager: React.FC<PoolManagerProps> = ({
...
+                    headers: { Authorization: localStorage.getItem('token') },
                 },
             )
             .then((response) => {
                 updatePool(response.data)
             })
             .catch((error) => {
                 setErrorAlert(
-                    `Sharing pool failed with error: ${error.message}`,
+                    `Sharing pool failed with error: ${error.response.data.detail}`,
```
Comment (EddieTheCubeHead) [inline client/components/poolmanagerComponents/poolManager.tsx:39]: "Sharing **a** pool"

File: client/components/buttons/iconButtons/showMoreIconButton.tsx [hunk 3]
```diff
@@ -44,15 +42,15 @@ const ShowMoreIconButton: React.FC<ShowMoreIconButtonProps> = ({
...
+                headers: { Authorization: localStorage.getItem('token') },
             })
             .then((response) => {
                 updatePool(response.data)
                 enableAddButton()
             })
             .catch((error) => {
                 setErrorAlert(
-                    `Creating pool failed with error: ${error.message}`,
+                    `Creating pool failed with error: ${error.response.data.detail}`,
```
Comment (EddieTheCubeHead) [inline client/components/buttons/iconButtons/showMoreIconButton.tsx:53]: "Creating **a** pool"

**Review by EddieTheCubeHead — CHANGES_REQUESTED** [review]
Super minor grammatical corrections, but should be done anyways imo.

**Review by EddieTheCubeHead — DISMISSED** [review]

**Comment by OttoAhlfors** [comment]
Joining a pool failed with error : [object Object]
There is an issue with the details following different structures depending on the error leading to slightly not helpfull error messages

**Review by tuomasmustakallio — DISMISSED** [review]

**Review by tuomasmustakallio — APPROVED** [review]

**Pull request merged**